src/components/Story/Story.js

```diff
@@ -81,36 +98,57 @@ class Story extends React.Component {
     let followText = '';
 
     if (postState.userFollowed && !pendingFollow) {
-      followText = intl.formatMessage({ id: 'unfollow_username', defaultMessage: 'Unfollow {username}' }, { username: post.author });
+      followText = intl.formatMessage(
+        { id: 'unfollow_username', defaultMessage: 'Unfollow {username}' },
+        { username: post.author },
+      );
     } else if (postState.userFollowed && pendingFollow) {
-      followText = intl.formatMessage({ id: 'unfollow_username', defaultMessage: 'Unfollow {username}' }, { username: post.author });
+      followText = intl.formatMessage(
+        { id: 'unfollow_username', defaultMessage: 'Unfollow {username}' },
+        { username: post.author },
+      );
     } else if (!postState.userFollowed && !pendingFollow) {
-      followText = intl.formatMessage({ id: 'follow_username', defaultMessage: 'Follow {username}' }, { username: post.author });
+      followText = intl.formatMessage(
+        { id: 'follow_username', defaultMessage: 'Follow {username}' },
+        { username: post.author },
+      );
     } else if (!postState.userFollowed && pendingFollow) {
-      followText = intl.formatMessage({ id: 'follow_username', defaultMessage: 'Follow {username}' }, { username: post.author });
+      followText = intl.formatMessage(
+        { id: 'follow_username', defaultMessage: 'Follow {username}' },
+        { username: post.author },
+      );
     }
 
     let popoverMenu = [];
 
     if (ownPost && post.cashout_time !== '1969-12-31T23:59:59') {
-      popoverMenu = [...popoverMenu, <PopoverMenuItem key="edit">
-        {saving ? <Icon type="loading" /> : <i className="iconfont icon-write" />}
-        <FormattedMessage id="edit_post" defaultMessage="Edit post" />
-      </PopoverMenuItem>];
+      popoverMenu = [
```

Comment:
Same thing here, looks like a lot of logic in this render function, it might be good to break it down into its own function maybe `this.renderPopoverMenu()`, also same with the rebloggedUI logic.

Reply:
Although I agree with this I don't think that it's related to this PR and working on multiple unrelated pieces in the same PR shouldn't be encouraged. I don't think we should rewrite half of our code base while adding one feature. I think we should work on this in new PR.